README.md

```diff
@@ -7,25 +7,33 @@ Experimental adjunct to the
 [![Build Status](https://travis-ci.org/srl295/gp-js-flatten.svg?branch=master)](https://travis-ci.org/srl295/gp-js-flatten)
 [![Coverage Status](https://coveralls.io/repos/github/srl295/gp-js-flatten/badge.svg)](https://coveralls.io/github/srl295/gp-js-flatten)
 
+Purpose
+=======
+
+The purpose of this utility is to flatten key/value pairs so that all keys and values are strings. Some types are ignored, such as empty right hand side values.
+
+Also simple keys and values are by default left alone, so `{key: "value"}=flatten({key:"value"})`.
+The `flattenAll: true` option will include these keys in the flattening.
 
 Usage
-===
+=====
 
 Given `input=`:
 
-```json
+<code>
```

Comment:
@yumaoka i changed it to `<code>` to allow the doc generation to pull in the sample json files directly, so that the doc stays exactly in sync with `input1.json` etc. I wasn't able to import it as ``` form.